src/BlazorWebView/src/Maui/Android/BlazorWebViewHandler.Android.cs

```diff
@@ -77,19 +74,25 @@ protected override void DisconnectHandler(AWebView platformView)
 					.DisposeAsync()
 					.AsTask()!;
 
-				if (IsAndroidFireAndForgetAsyncEnabled)
-				{
-					// If the app is configured to fire-and-forget via an AppContext Switch, we'll do that.
-					disposalTask.FireAndForget();
-				}
-				else
+				// When determining whether to block on disposal, we respect the more specific AndroidFireAndForgetAsync switch
+				// if specified. If not, we fall back to the general UseBlockingDisposal switch, defaulting to false.
+				var shouldBlockOnDispose = AppContext.TryGetSwitch(AndroidFireAndForgetAsyncSwitch, out var enableFireAndForget)
+					? !enableFireAndForget
+					: IsBlockingDisposalEnabled;
```

Comment:
This logic ensures that if the `"BlazorWebView.AndroidFireAndForgetAsync"` switch was specified previously, we will continue to use that value, regardless of whether the new `AppContext` switch is also specified.